xcore/interface/stitcher.cpp

```diff
@@ -629,22 +629,39 @@ BowlModel::get_bowlview_vertex_model (
                 bowl_view_image_to_world (
                     _config, _bowl_img_width, _bowl_img_height, texture_pos0);
 
-            vertices.push_back (PointFloat3(world_pos0.x / _config.a, world_pos0.y / _config.b, world_pos0.z / _config.c));
-            indeices.push_back (indicator++);
-            texture_points.push_back (PointFloat2(texture_pos0.x / _bowl_img_width, (_bowl_img_height - texture_pos0.y) / _bowl_img_height));
+            if (world_pos0.z / _config.c < view_height) {
+                if (top_view) {
+                    world_pos0.z = 0.0f;
```

Comment:
top_view is not just set world_pos.z to 0.0f.  If so, this can cause lots useless vertices into calculation. try best to keep precise calculation.
the start point of ground height can be found in 
https://github.com/01org/libxcam/blob/24b1d88ea25117db2d923bbf9114897397bb25c0/xcore/xcam_utils.cpp#L98

Since top-view and bowl-view code are same except height start position. You can extract them into a single function.